hpke/shortkem.go

```diff
@@ -0,0 +1,142 @@
+package hpke
+
+import (
+	"crypto/elliptic"
+	"crypto/rand"
+	_ "crypto/sha256" // linking sha256 packages.
+	_ "crypto/sha512" // linking sha512 packages.
+	"crypto/subtle"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/cloudflare/circl/kem"
+)
+
+type shortKem struct {
+	kemBase
+	elliptic.Curve
+}
+
+func (s shortKem) PrivateKeySize() int        { return s.byteSize() }
+func (s shortKem) SeedSize() int              { return s.byteSize() }
+func (s shortKem) CiphertextSize() int        { return 1 + 2*s.byteSize() }
+func (s shortKem) PublicKeySize() int         { return 1 + 2*s.byteSize() }
+func (s shortKem) EncapsulationSeedSize() int { return s.byteSize() }
+
+func (s shortKem) byteSize() int { return (s.Params().BitSize + 7) / 8 }
+
+func (s shortKem) sizeDH() int { return s.byteSize() }
+func (s shortKem) calcDH(dh []byte, sk kem.PrivateKey, pk kem.PublicKey) error {
+	PK := pk.(*shortPubKey)
+	SK := sk.(*shortPrivKey)
+	l := len(dh)
+	x, _ := s.ScalarMult(PK.x, PK.y, SK.k) // only x-coordinate is used.
+	b := x.Bytes()
+	copy(dh[l-len(b):l], b)
+	return nil
+}
+
+func (s shortKem) DeriveKey(seed []byte) (kem.PublicKey, kem.PrivateKey) {
```

Comment:
This derives a pair of keys, not a key.

```suggestion
func (s shortKem) DeriveKeyPair(seed []byte) (kem.PublicKey, kem.PrivateKey) {
```